library/entropy_poll.c

```diff
@@ -29,7 +29,7 @@
 
 #if !defined(unix) && !defined(__unix__) && !defined(__unix) && \
     !defined(__APPLE__) && !defined(_WIN32) && !defined(__QNXNTO__) && \
-    !defined(__HAIKU__) && !defined(__midipix__)
+    !defined(__HAIKU__) && !defined(__midipix__) && !defined(__MVS__)
```

Comment:
Going by [IBM documentation](https://www.ibm.com/docs/en/zos/2.1.0?topic=macros-related-platform) I would have expected `__TOS_MVS__`, since what matters is the target you're compiling for, not the host you're building on. Or do “host” and “target” mean different things here?

Reply:
In this case, host _is_ the target. Can't build for z/OS outside of z/OS currently.
IBM is working on adding the platform's support upstream (LLVM), working to provide platform access to open source projects as required, and to a compiler on the platform (https://www.ibm.com/docs/en/cpp-oel-zos/2.0).
Almost all the patches sent upstream (via https://github.com/ZOSOpenTools) use `__MVS__`.

Reply:
Ok. What about building on z/OS for some other OS? (I'm not sure if anybody cares about that.) What macros would be defined then?

Reply:
Highly unlikely that such a thing will happen :sweat_smile:.
Anyway, I don't know if that is even possible, or has been done (I'm not an IBMer BTW, just a passerby mainframer).